android/src/main/java/io/mosip/tuvali/openid4vpble/VerifierModule.kt

```diff
@@ -0,0 +1,96 @@
+package io.mosip.tuvali.openid4vpble
+
+import android.util.Log
+import com.facebook.react.bridge.*
+import io.mosip.tuvali.common.safeExecute.TryExecuteSync
+import io.mosip.tuvali.common.uri.OpenId4vpURI
+import io.mosip.tuvali.exception.handlers.OpenIdBLEExceptionHandler
+import io.mosip.tuvali.openid4vpble.events.withArgs.VerificationStatusEvent
+import io.mosip.tuvali.openid4vpble.events.withoutArgs.DisconnectedEvent
+import io.mosip.tuvali.transfer.Util.Companion.getLogTag
+import io.mosip.tuvali.verifier.Verifier
+
+class VerifierModule(private val reactContext: ReactApplicationContext) :
+  ReactContextBaseJavaModule(reactContext) {
+
+  private val eventEmitter = EventEmitter(reactContext)
+  private val logTag = getLogTag(javaClass.simpleName)
+  private var verifier: Verifier? = null
+  private var bleExceptionHandler = OpenIdBLEExceptionHandler(eventEmitter::emitError, this::stopBLE)
+  private val tryExecuteSync = TryExecuteSync(bleExceptionHandler)
+
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  fun setTuvaliVersion(version: String) {
+    tuvaliVersion = version
+  }
+
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  fun startAdvertisement(advIdentifier: String): String {
+    Log.d(logTag, "startAdvertisement called with advIdentifier $advIdentifier at ${System.nanoTime()}")
+
+    return tryExecuteSync.run {
+      if (verifier == null) {
+        initializeVerifier()
+      }
+
+      val payload = verifier!!.getAdvPayloadInHex(advIdentifier)
+      Log.d(logTag, "synchronized startAdvertisement called with adv identifier $payload at ${System.nanoTime()} and verifier hashcode: ${verifier.hashCode()}")
+
+      verifier?.startAdvertisement(advIdentifier)
+
+      return@run OpenId4vpURI(payload).toString()
```

Comment:
Constructor can take multiple params like advIdentifier, publicKey and construct the URI internally.

Reply:
updated